src/handlers/expiry/providers/TestDataProvider.ts

```diff
@@ -197,7 +198,60 @@ export class TestDataProvider implements ITestDataProvider {
     return testResults;
   }
 
-  public static filterTestResultsByParam(
+  public async getTestTypesWithTestCodesAndClassification(
+    testTypes: models.TestType[] = [],
+    testTypeParams: models.TestTypeParams
+  ) {
+    testTypes.forEach(async (testType) => {
+      this.testResultsDAO = this.testResultsDAO as models.TestResultsDAO;
+      const { testTypeId } = testType;
+      const {
+        defaultTestCode,
+        linkedTestCode,
+        testTypeClassification,
+      } = await this.testResultsDAO.getTestCodesAndClassificationFromTestTypes(
+        testTypeId,
+        testTypeParams
+      );
+      testType.testCode =
+        testTypes.length > 1
+          ? linkedTestCode
```

Comment:
do we need this nested conditions here?
It seems that the default value should always be defaultTestCode ?

Reply:
cleaned up the conditions.